### PR TITLE
Cleanup job for deleted attachments

### PR DESCRIPTION
**Pull request by IonutPad15**

### CODE REVIEW

File: IssueTracker/DataAccess/Repository/FileRepository.cs [hunk 1]
```diff
@@ -10,24 +10,34 @@ public FileRepository(ISQLDataAccess db)
     }
     public async Task<string?> AddAsync(Models.File entity)
     {
+        DateTime upate = DateTime.UtcNow;
```
Comment (iiusan) [inline IssueTracker/DataAccess/Repository/FileRepository.cs:13]: typo

File: IssueTracker/DataAccess/Repository/FileRepository.cs [hunk 2]
```diff
@@ -10,24 +10,34 @@ public FileRepository(ISQLDataAccess db)
...
+        await _db.SaveDataAsync("dbo.spFile_Delete", new { FileId = fileId, Updated = upate });
     }
 
     public async Task<IEnumerable<Models.File>> GetByIssueIdAsync(int issueId)
     {
         var result = await _db.LoadDataAsync<Models.File, dynamic>("spFile_GetByIssueId", new { IssueId = issueId });
         return result;
     }
+    public async Task<IEnumerable<Models.File>> GetForCleanupAsync(int days)
```
Comment (iiusan) [inline IssueTracker/DataAccess/Repository/FileRepository.cs]: Should be more generic. Use TimeSpan

File: IssueTracker/IssueTracker.FileSystem/Repository/FileProvider.cs [hunk 3]
```diff
@@ -1,5 +1,7 @@
-﻿using Microsoft.Extensions.Configuration;
+﻿using System.Runtime.CompilerServices;
+using Microsoft.Extensions.Configuration;
 
+[assembly: InternalsVisibleTo("IssueTracker.UnitTest")]
```
Comment (iiusan) [inline IssueTracker/IssueTracker.FileSystem/Repository/FileProvider.cs:4]: You need this only once.

File: IssueTracker/IssueTracker.FileSystem/Repository/IRepository/IBolbData.cs [hunk 4]
```diff
@@ -1,7 +1,10 @@
...
+﻿using System.Runtime.CompilerServices;
+
+[assembly: InternalsVisibleTo("IssueTracker.UnitTest")]
+namespace IssueTracker.FileSystem;
 internal interface IBolbData
 {
     Task<IEnumerable<Models.File>> GetFilesAsync(IEnumerable<Models.File> files);
     Task UploadFileAsync(Models.File file);
-    Task<Stream> DownloadFileAsync(string link);
+    Task<bool> DeleteFileAsync(string name);
```
Comment (iiusan) [inline IssueTracker/IssueTracker.FileSystem/Repository/IRepository/IBolbData.cs]: The name can be simplified. DeleteAsync()

File: IssueTracker/IssueTracker.FileSystem/Repository/IRepository/IBolbData.cs [hunk 5]
```diff
@@ -1,7 +1,10 @@
-﻿namespace IssueTracker.FileSystem;
+﻿using System.Runtime.CompilerServices;
+
+[assembly: InternalsVisibleTo("IssueTracker.UnitTest")]
+namespace IssueTracker.FileSystem;
 internal interface IBolbData
```
Comment (iiusan) [inline IssueTracker/IssueTracker.FileSystem/Repository/IRepository/IBolbData.cs]: Consistency! Rename to IBlobProvider

File: IssueTracker/IssueTracker.FileSystem/Repository/IRepository/IFileProvider.cs [hunk 6]
```diff
@@ -3,5 +3,5 @@ public interface IFileProvider
 {
     Task UploadAsync(Models.File file);
```
Comment (iiusan) [inline IssueTracker/IssueTracker.FileSystem/Repository/IRepository/IFileProvider.cs:4]: are Models.File and IssueTracker.FileSystem.Models.File the same dto?

File: IssueTracker/IssueTracker.FileSystem/Repository/MetaData.cs [hunk 7]
```diff
@@ -15,8 +17,7 @@ internal MetaData(IMetaDataConfiguration config)
     }
     public IEnumerable<Models.File> GetAll(IEnumerable<Models.File> files)
```
Comment (iiusan) [inline IssueTracker/IssueTracker.FileSystem/Repository/MetaData.cs:16]: Do you need GetAll ?

File: IssueTracker/IssueTracker.FileSystem/Repository/MetaData.cs [hunk 8]
```diff
@@ -64,7 +64,7 @@ public async Task<bool> DeleteAsync(string id, string group)
         var operation = TableOperation.Retrieve<MetaDataEntity>(group, id);
         var result = await _metaDataTable.ExecuteAsync(operation);
         var x = result.Result as MetaDataEntity;
-        if (x != null && x.IsDeleted == false)
+        if (x != null)
```
Comment (iiusan) [inline IssueTracker/IssueTracker.FileSystem/Repository/MetaData.cs]: x? rename

File: IssueTracker/IssueTrackerAPI/Controllers/FileController.cs [hunk 9]
```diff
@@ -86,9 +87,14 @@ public async Task<IResult> Delete([FromBody] FileDeleteRequest fileDelete)
             List<ValidationFailure> failures = results.Errors;
             return Results.BadRequest(failures);
         }
-        await _fileData.DeleteAsync(fileDelete.FileId);
-        var file = _mapper.Map<IssueTracker.FileSystem.Models.File>(fileDelete);
-        await _fileProvider.DeleteAsync(file);
+        try
+        {
+            await _fileData.DeleteAsync(fileDelete.FileId);
```
Comment (iiusan) [inline IssueTracker/IssueTrackerAPI/Controllers/FileController.cs]: rename _fileData

File: IssueTracker/IssueTrackerAPI/appsettings.json [hunk 10]
```diff
@@ -12,7 +12,8 @@
     "AccountName": "devstoreaccount1",
     "AccountKey": "Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==",
     "Container": "azureblobstorage",
-    "AzureTable": "MetaData"
+    "AzureTable": "MetaData",
+    "CleanupDaysInterval": "2"
```
Comment (iiusan) [inline IssueTracker/IssueTrackerAPI/appsettings.json]: should be a TimeSpan

**Review by iiusan — CHANGES_REQUESTED** [review]

File: IssueTracker/DataAccess/Repository/FileRepository.cs (hunk 1 above)
Comment (IonutPad15) [reply IssueTracker/DataAccess/Repository/FileRepository.cs:13]: changed

File: IssueTracker/DataAccess/Repository/FileRepository.cs (hunk 2 above)
Comment (IonutPad15) [reply IssueTracker/DataAccess/Repository/FileRepository.cs]: ok

File: IssueTracker/IssueTracker.FileSystem/Repository/FileProvider.cs (hunk 3 above)
Comment (IonutPad15) [reply IssueTracker/IssueTracker.FileSystem/Repository/FileProvider.cs:4]: done

File: IssueTracker/IssueTracker.FileSystem/Repository/IRepository/IBolbData.cs (hunk 4 above)
Comment (IonutPad15) [reply IssueTracker/IssueTracker.FileSystem/Repository/IRepository/IBolbData.cs]: done

File: IssueTracker/IssueTracker.FileSystem/Repository/IRepository/IBolbData.cs (hunk 5 above)
Comment (IonutPad15) [reply IssueTracker/IssueTracker.FileSystem/Repository/IRepository/IBolbData.cs]: done

File: IssueTracker/IssueTracker.FileSystem/Repository/IRepository/IFileProvider.cs (hunk 6 above)
Comment (IonutPad15) [reply IssueTracker/IssueTracker.FileSystem/Repository/IRepository/IFileProvider.cs:4]: no

File: IssueTracker/IssueTracker.FileSystem/Repository/MetaData.cs (hunk 7 above)
Comment (IonutPad15) [reply IssueTracker/IssueTracker.FileSystem/Repository/MetaData.cs:16]: Claudiu used it in GetAsync

File: IssueTracker/IssueTracker.FileSystem/Repository/MetaData.cs (hunk 8 above)
Comment (IonutPad15) [reply IssueTracker/IssueTracker.FileSystem/Repository/MetaData.cs]: ok

File: IssueTracker/IssueTrackerAPI/Controllers/FileController.cs (hunk 9 above)
Comment (IonutPad15) [reply IssueTracker/IssueTrackerAPI/Controllers/FileController.cs]: done

File: IssueTracker/IssueTrackerAPI/appsettings.json (hunk 10 above)
Comment (IonutPad15) [reply IssueTracker/IssueTrackerAPI/appsettings.json]: ok

**Review by iiusan — APPROVED** [review]

**Pull request merged**